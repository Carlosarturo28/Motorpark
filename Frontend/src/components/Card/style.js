import styled from 'styled-components'
import { Link } from 'react-router-dom';

export const CardContainer = styled.button`
background-color: #fff;
cursor: pointer;
border-width: 0;
width: 100%;
font-family: inherit;
border-radius: 8px;
padding: 12px 8px;
margin: 10px 0;
text-align: left;
display: flex;
-webkit-box-shadow: 0px 11px 23px -5px rgba(0,107,78,1);
-moz-box-shadow: 0px 11px 23px -5px rgba(0,107,78,1);
box-shadow: 0px 11px 23px -5px rgba(0,107,78,1);
`

export const LinkTo = styled(Link)`
text-decoration: none;
`

export const CarImage = styled.img`
height: 14vh;
`

export const InfoContainer = styled.div`
flex-direction: column;
display: flex;
align-self: center;
padding: 0 10px;
`

export const BottomCard = styled.div`
display: flex;
flex-direction: row;
justify-content: flex-start;
flex-wrap: wrap;
`

export const Brand = styled.p`
font-size: 16px;
font-weight: 700;
margin: 0 0 3px 0;
justify-content: left;
color: #777;
`

export const Model = styled.p`
font-size: 16px;
font-weight: 500;
margin: 0 0 3px 4px;
justify-content: left;
color: #777;
`

export const Year = styled.p`
font-size: 14px;
font-weight: 400;
margin: 3px 0 3px 0;
color: #777;
`

export const Plate = styled.p`
font-size: 14px;
font-weight: 400;
margin: 3px 0 0 0;
justify-content: left;
color: #777;
`

export const Color = styled.p`
font-size: 14px;
font-weight: 400;
margin: 3px 0 3px 0;
justify-content: left;
color: #777;
`