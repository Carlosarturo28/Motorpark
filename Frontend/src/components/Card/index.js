import React from 'react'
import PropTypes from 'prop-types'
import { CardContainer, Brand, Model, LinkTo, Year, Color, Plate, BottomCard, CarImage, InfoContainer } from './style'

const Card = ({ brand, model, year, color, item, plate, onPress }) => (
<LinkTo to={{
  pathname: '/edit-vehicle',
  state: {
    item: item
  }
}}>
<CardContainer>
<CarImage src={'https://images-na.ssl-images-amazon.com/images/I/41Y%2B5ZV5TOL._SS600_.jpg'} />
<InfoContainer>
<BottomCard>
<Brand>{brand}</Brand>
<Model>{model}</Model>
</BottomCard>
<Year>Year: {year}</Year>
<Color>Color: {color}</Color>
<Plate>Plate: {plate.toUpperCase()}</Plate>
</InfoContainer>
</CardContainer>
</LinkTo>
)

Card.defaultProps = {
brand: 'Renault', 
model: 'Sandero',
year: 2019,
color: 'red',
plate: 'isx555'
}

Card.propTypes = {
  brand: PropTypes.string,
  model: PropTypes.string,
  year: PropTypes.number,
  color: PropTypes.string,
  plate: PropTypes.string
}

export default Card