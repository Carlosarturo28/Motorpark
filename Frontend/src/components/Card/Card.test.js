import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';

describe('Card', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Card debug />);
  
    expect(component).toMatchSnapshot();
  });

it('should render card correctly with given props', () => {
  const props = { brand: 'Kia', model: 'Cerato', year: 2019, color: 'black', plate: 'TST002' };
  const component = shallow(<Card brand={props.brand} model={props.model} year={props.year} color={props.color} plate={props.plate} />);
  expect(component).toMatchSnapshot();
});

});