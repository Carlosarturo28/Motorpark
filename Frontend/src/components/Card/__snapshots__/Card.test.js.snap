// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render card correctly with given props 1`] = `
<Styled(Link)
  to={
    Object {
      "pathname": "/edit-vehicle",
      "state": Object {
        "item": undefined,
      },
    }
  }
>
  <styled.button>
    <styled.img
      src="https://images-na.ssl-images-amazon.com/images/I/41Y%2B5ZV5TOL._SS600_.jpg"
    />
    <styled.div>
      <styled.div>
        <styled.p>
          Kia
        </styled.p>
        <styled.p>
          Cerato
        </styled.p>
      </styled.div>
      <styled.p>
        Year: 
        2019
      </styled.p>
      <styled.p>
        Color: 
        black
      </styled.p>
      <styled.p>
        Plate: 
        TST002
      </styled.p>
    </styled.div>
  </styled.button>
</Styled(Link)>
`;

exports[`Card should render correctly in "debug" mode 1`] = `
<Styled(Link)
  to={
    Object {
      "pathname": "/edit-vehicle",
      "state": Object {
        "item": undefined,
      },
    }
  }
>
  <styled.button>
    <styled.img
      src="https://images-na.ssl-images-amazon.com/images/I/41Y%2B5ZV5TOL._SS600_.jpg"
    />
    <styled.div>
      <styled.div>
        <styled.p>
          Renault
        </styled.p>
        <styled.p>
          Sandero
        </styled.p>
      </styled.div>
      <styled.p>
        Year: 
        2019
      </styled.p>
      <styled.p>
        Color: 
        red
      </styled.p>
      <styled.p>
        Plate: 
        ISX555
      </styled.p>
    </styled.div>
  </styled.button>
</Styled(Link)>
`;
